react/features/app/components/AbstractApp.js

```diff
@@ -133,26 +133,14 @@ export class AbstractApp extends Component {
             // actions is important, not the call site. Moreover, we've got
             // localParticipant business logic in the React Component
             // (i.e. UI) AbstractApp now.
-            let localParticipant = {};
-
-            if (typeof APP === 'object') {
-                localParticipant = {
-                    avatarID: APP.settings.getAvatarId(),
-                    avatarURL: APP.settings.getAvatarUrl(),
-                    email: APP.settings.getEmail(),
-                    name: APP.settings.getDisplayName()
-                };
-            }
-
-            // Profile is the new React compatible settings.
-            const profile = getState()['features/base/profile'];
 
-            if (profile) {
-                localParticipant.email
-                    = profile.email || localParticipant.email;
-                localParticipant.name
-                    = profile.displayName || localParticipant.name;
-            }
+            const settings = getSettings(getState());
```

Comment:
Now that the logic is simplified, can you address the comment above and move this to a middleware in base/parcipants?

Reply:
Ok, done!